imcsdk/apis/server/storage.py

```diff
@@ -270,41 +293,726 @@ def vd_query_by_name(handle,
+                handle,
+                controller_type='SAS',
+                controller_slot='HBA'',
+                drive_slot=4')
+    """
+    drive = physical_drive_get(handle, controller_type,
+                               controller_slot, drive_slot,
+                               server_id)
+    if drive is None:
+        raise ImcOperationError("Get Physical Drive:%s" % drive_slot,
+                                "Managed Object not found")
+    return drive.fde_enabled.lower() in ['yes', 'true']
+
+
+def physical_drive_set_jbod_mode(handle, controller_type,
+                                 controller_slot, drive_slot, server_id=1):
+    """
+    Sets the physical drive in jbod mode
+
+    Args:
+        handle (ImcHandle)
+        controller_type (str): Controller type
+                               'SAS'
+        controller_slot (str): Controller slot name/number
+                                "MEZZ","0"-"9", "HBA"
+        drive_slot (int): Slot in which the drive resides
+        server_id (int): server_id for UCS 3260 platform
+
+    Returns:
+        StorageLocalDisk object
+
+    Examples:
+        physical_drive_set_jbod_mode(handle,
+                                 controller_type='SAS',
+                                 controller_slot='HBA'',
+                                 drive_slot=4')
+    """
+    if not is_controller_jbod_mode_enabled(handle, controller_type,
+                                           controller_slot, server_id):
+        raise ImcOperationError("Physical Drive: %s JBOD Mode Enable",
+                                "Controller JBOD mode is not enabled")
+    pd = physical_drive_get(handle, controller_type, controller_slot,
+                            drive_slot, server_id)
+    return _set_physical_drive_action(handle,
+                                      pd,
+                                      StorageLocalDiskConsts.ADMIN_ACTION_MAKE_JBOD)
+
+
+def physical_drive_set_unconfigured_good(handle,
+                                         controller_type,
+                                         controller_slot,
+                                         drive_slot,
+                                         server_id=1):
+    """
+    Sets the physical drive in unconfigured good state
+
+    Args:
+        handle (ImcHandle)
+        controller_type (str): Controller type
+                               'SAS'
+        controller_slot (str): Controller slot name/number
+                                "MEZZ","0"-"9", "HBA"
+        drive_slot (int): Slot in which the drive resides
+        server_id (int): server_id for UCS 3260 platform
+
+    Returns:
+        StorageLocalDisk object
+
+    Examples:
+        physical_drive_set_unconfigured_good(
+                handle,
+                controller_type='SAS',
+                controller_slot='HBA'',
+                drive_slot=4')
+    """
+    pd = physical_drive_get(handle, controller_type, controller_slot,
```

Comment:
most of the code here is repeated for every method. just to change a flag..
having multiple user facing APIs is good. But internally we could wrap all this in a single internal API.  will save us a lot of code.